WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/data/db/AnalogWatchFaceDao.kt

```diff
@@ -29,10 +30,14 @@ import kotlinx.coroutines.flow.Flow
 @Dao
 interface AnalogWatchFaceDao {
     @Query("SELECT * FROM analog_watch_face_table ORDER BY name ASC")
-    fun getAll(): Flow<List<AnalogWatchFaceEntity>>
+    suspend fun getAll(): List<AnalogWatchFaceEntity>
 
-    @Query("SELECT * FROM analog_watch_face_table WHERE id=(:id)")
-    fun get(id: Int): Flow<AnalogWatchFaceEntity>
+    @Query("SELECT * FROM analog_watch_face_table WHERE id=(:id) LIMIT 1")
+    suspend fun get(id: Int): AnalogWatchFaceEntity
+
+    @Transaction
+    @Query("SELECT * FROM analog_watch_face_table WHERE id=(:id) LIMIT 1")
+    fun getAnalogWatchFaceAndStylesAndDimensions(id: Int): Flow<AnalogWatchFaceAndStylesAndDimensions>
```

Comment:
Maybe use `getWithStylesAndDimensions`? You leave out the AnalogWatchFace part for the other methods as well.

Reply:
Good suggestion!